gcp/plugin.go

```diff
@@ -28,16 +28,20 @@ func Plugin(ctx context.Context) *plugin.Plugin {
 		DefaultIgnoreConfig: &plugin.IgnoreConfig{
 			ShouldIgnoreErrorFunc: shouldIgnoreErrorPluginDefault(),
 		},
-		ConnectionKeyColumns: []plugin.ConnectionKeyColumn{
+		RateLimiters: []*rate_limiter.Definition{
+			// A user can make up to 100 BigQuery API requests per second, per user, per project to an API method. If a user makes more than 100 requests per second to a method, then throttling can occur. This limit does not apply to
+			// Service API limit: https://cloud.google.com/bigquery/quotas#api_request_quotas
+			// https://cloud.google.com/bigquery/quotas#dataset_limits
+			// Configuring the rate limiter only for LIST operations will result in hitting rate limits when executing the query "select * from gcp_bigquery_dataset".
+			// This occurs because the table `gcp_bigquery_dataset` includes hydrated calls (getBigQueryDataset). Therefore, it is necessary to configure the rate limiter for both LIST and GET operations.
 			{
-				Name:    "project",
-				Hydrate: getProject,
+				Name:           "gcp_bigquery_list_datasets",
```

Comment:
If we keep this combined, we should use the name `gcp_bigquery_list_get_datasets` to represent we're covering both operations